Add hive config flow

KJonline opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Add Hive config flow to allow hive to be setup via the UI.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #43890
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/16568

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

Commit:
hive config flow

Commit:
hive config flow

Commit:
hive config flow

Commit:
hive config flow

KJonline commented:
@MartinHjelmare This is PR number 3.

yuvalabou commented on homeassistant/components/hive/strings.json:
```diff
@@ -0,0 +1,46 @@
+        "description": "Enter your Hive login information and configuration.",
+        "data": {
+          "username": "Email Address",
+          "password": "Password",
```
You can use global keys from here - https://github.com/home-assistant/core/blob/dev/homeassistant/strings.json

Commit:
update to Bool check

Commit:
update to Bool check

Commit:
update to Bool check

Commit:
Updated Strings.json

Commit:
Updated Strings.json

Commit:
Updated Strings.json

Commit:
Updated Strings.json

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -69,75 +64,113 @@
+async def async_setup_entry(hass, entry):
+    """Set up Hive from a config entry."""
+    # Store an API object for your platforms to access
+    # hass.data[DOMAIN][entry.entry_id] = MyApi(...)
```
Commented code.

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -69,75 +64,113 @@
 
         if mode == "on":
-            hive.hotwater.turn_boost_on(hive_id, minutes)
+            await hive.hotwater.turn_boost_on(device, minutes)
```
How did this work before?

KJonline replied:
it used to store the hive_id and pass that to the pyhiveapi library but now it stores the device against the entry_id and passed the to pyhiveapi.

MartinHjelmare replied:
We haven't updated the library. How can this api change without a library update?

KJonline replied:
Unfortunately this was an oversight in the last version, but is fixed now.

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -69,75 +64,113 @@
-    config["update_interval"] = config[DOMAIN][CONF_SCAN_INTERVAL]
-
-    devices = await hive.session.startSession(config)
+    hive_options = hive_options if len(hive_options) > 0 else hive_config["options"]
```
```suggestion
    hive_options = hive_options if hive_options else hive_config["options"]
```

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -69,75 +64,113 @@
-    devices = await hive.session.startSession(config)
+    hive_options = hive_options if len(hive_options) > 0 else hive_config["options"]
+    hive_config["options"].update(hive_options)
+    hive_config["add_sensors"] = bool("hive" in hass.data["custom_components"])
```
Leftover from testing?

KJonline replied:
No this line needs to stay in as its how we will identify and tell pyhiveapi if someone is using a custom version of the integration and will setup extra sensors. similar login used to live in the sensors file but has been moved to here now.

MartinHjelmare replied:
We don't allow that. It needs to be removed.

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -69,75 +64,113 @@
+    hive_config["options"].update(hive_options)
+    hive_config["add_sensors"] = bool("hive" in hass.data["custom_components"])
+    hass.config_entries.async_update_entry(entry, options=hive_options)
+    hass.data[DOMAIN][entry.entry_id] = hive
```
Probably nest the entry ids under a specific key and sub dict so we know what we're trying to look up.

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -69,75 +64,113 @@
-        _LOGGER.error("Hive API initialization failed")
-        return False
+    try:
+        hive.devices = await hive.session.startSession(hive_config)
```
Why are we modifying the `Hive` instance? I'd expect the `Hive` class to be responsible for that.

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -69,75 +64,113 @@
-    for ha_type in DEVICETYPES:
-        devicelist = devices.get(DEVICETYPES[ha_type])
+    for component in PLATFORMS:
+        devicelist = hive.devices.get(component)
```
There needs to be a decoupling between the hive names and the home assistant platform names. Please define a constant dict that maps these.

KJonline replied:
The dictionary has keys the same as the home assistant platform names, so the dictionary would be.

```python
LOOKUP = {
"binary_sensor": "binary_sensor",
"climate": "climate",
"light": "light",
"sensor": "sensor",
"switch": "switch",
"water_heater": "water_heater"
}
```

This seems a but overkill to loop through a dictionary for the same value. Im happy to write a comment explaining this so others are aware.

MartinHjelmare replied:
No we need this decoupling to make sure that we don't break silently if any of the two APIs change.

MartinHjelmare commented on homeassistant/components/hive/binary_sensor.py:
```diff
@@ -25,11 +26,16 @@
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```
We can remove this.

KJonline replied:
Do you mean the whole function or just the parameters?

MartinHjelmare replied:
The whole function.

MartinHjelmare reviewed (changes requested):
Please add tests for the config flow. Then I'll review that module too.

davet2001 commented:
I have tested the branch on my hive system.

Firstly it seems to work well!  I logged in (without 2FA enabled), and it connected, identified my hub and thermostat, then allowed me to control heating settings.  

When I changed settings on my hive app, they updated in HA (within ~2mins), and when I updated on HA, they updated on the app.

During startup I got this though:
````
2021-02-15 20:53:48 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 118, in impl
    return await handler(request)
  File "/workspaces/core/homeassistant/components/http/security_filter.py", line 56, in security_filter_middleware
    return await handler(request)
  File "/workspaces/core/homeassistant/components/http/request_context.py", line 18, in request_context_middleware
    return await handler(request)
  File "/workspaces/core/homeassistant/components/http/ban.py", line 72, in ban_middleware
    return await handler(request)
  File "/workspaces/core/homeassistant/components/http/auth.py", line 129, in auth_middleware
    return await handler(request)
  File "/workspaces/core/homeassistant/components/http/view.py", line 129, in handle
    result = await result
  File "/workspaces/core/homeassistant/components/config/config_entries.py", line 169, in post
    return await super().post(request, flow_id)
  File "/workspaces/core/homeassistant/components/http/data_validator.py", line 60, in wrapper
    result = await method(view, request, *args, **kwargs)
  File "/workspaces/core/homeassistant/helpers/data_entry_flow.py", line 106, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/workspaces/core/homeassistant/data_entry_flow.py", line 158, in async_configure
    result = await self._async_handle_step(flow, cur_step["step_id"], user_input)
  File "/workspaces/core/homeassistant/data_entry_flow.py", line 216, in _async_handle_step
    result: Dict = await getattr(flow, method)(user_input)
  File "/workspaces/core/homeassistant/components/hive/config_flow.py", line 83, in async_step_user
    if self.tokens.get("ChallengeName") == "SMS_MFA":
AttributeError: 'str' object has no attribute 'get'
2021-02-15 20:54:04 INFO (MainThread) [homeassistant.setup] Setting up hive
2021-02-15 20:54:04 INFO (MainThread) [homeassistant.setup] Setup of domain hive took 0.0 seconds
2021-02-15 20:54:07 INFO (SyncWorker_7) [homeassistant.loader] Loaded climate from homeassistant.components.climate
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up binary_sensor.hive
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setting up climate
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setup of domain climate took 0.0 seconds
2021-02-15 20:54:07 INFO (SyncWorker_4) [homeassistant.loader] Loaded sensor from homeassistant.components.sensor
2021-02-15 20:54:07 INFO (SyncWorker_3) [homeassistant.loader] Loaded water_heater from homeassistant.components.water_heater
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.climate] Setting up climate.hive
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setting up sensor
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setup of domain sensor took 0.0 seconds
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setting up water_heater
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setup of domain water_heater took 0.0 seconds
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.hive
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new binary_sensor.hive entity: binary_sensor.hive_hub_status
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.water_heater] Setting up water_heater.hive
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new climate.hive entity: climate.your_thermostat
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new sensor.hive entity: sensor.your_thermostat_battery_level
2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new water_heater.hive entity: water_heater.hot_water
````
Also tried turning on 2FA and that worked too.  It prompted me to enter the number that had just appeared on my phone, it completed successfully, and bidirectional heating control worked fine.

So in summary it seems to work perfectly apart from the above error while installing the integration.

KJonline commented:
> I have tested the branch on my hive system.
> 
> Firstly it seems to work well! I logged in (without 2FA enabled), and it connected, identified my hub and thermostat, then allowed me to control heating settings.
> 
> When I changed settings on my hive app, they updated in HA (within ~2mins), and when I updated on HA, they updated on the app.
> 
> During startup I got this though:
> 
> ```
> 2021-02-15 20:53:48 ERROR (MainThread) [aiohttp.server] Error handling request
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
>     resp = await self._request_handler(request)
>   File "/usr/local/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
>     resp = await handler(request)
>   File "/usr/local/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 118, in impl
>     return await handler(request)
>   File "/workspaces/core/homeassistant/components/http/security_filter.py", line 56, in security_filter_middleware
>     return await handler(request)
>   File "/workspaces/core/homeassistant/components/http/request_context.py", line 18, in request_context_middleware
>     return await handler(request)
>   File "/workspaces/core/homeassistant/components/http/ban.py", line 72, in ban_middleware
>     return await handler(request)
>   File "/workspaces/core/homeassistant/components/http/auth.py", line 129, in auth_middleware
>     return await handler(request)
>   File "/workspaces/core/homeassistant/components/http/view.py", line 129, in handle
>     result = await result
>   File "/workspaces/core/homeassistant/components/config/config_entries.py", line 169, in post
>     return await super().post(request, flow_id)
>   File "/workspaces/core/homeassistant/components/http/data_validator.py", line 60, in wrapper
>     result = await method(view, request, *args, **kwargs)
>   File "/workspaces/core/homeassistant/helpers/data_entry_flow.py", line 106, in post
>     result = await self._flow_mgr.async_configure(flow_id, data)
>   File "/workspaces/core/homeassistant/data_entry_flow.py", line 158, in async_configure
>     result = await self._async_handle_step(flow, cur_step["step_id"], user_input)
>   File "/workspaces/core/homeassistant/data_entry_flow.py", line 216, in _async_handle_step
>     result: Dict = await getattr(flow, method)(user_input)
>   File "/workspaces/core/homeassistant/components/hive/config_flow.py", line 83, in async_step_user
>     if self.tokens.get("ChallengeName") == "SMS_MFA":
> AttributeError: 'str' object has no attribute 'get'
> 2021-02-15 20:54:04 INFO (MainThread) [homeassistant.setup] Setting up hive
> 2021-02-15 20:54:04 INFO (MainThread) [homeassistant.setup] Setup of domain hive took 0.0 seconds
> 2021-02-15 20:54:07 INFO (SyncWorker_7) [homeassistant.loader] Loaded climate from homeassistant.components.climate
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up binary_sensor.hive
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setting up climate
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setup of domain climate took 0.0 seconds
> 2021-02-15 20:54:07 INFO (SyncWorker_4) [homeassistant.loader] Loaded sensor from homeassistant.components.sensor
> 2021-02-15 20:54:07 INFO (SyncWorker_3) [homeassistant.loader] Loaded water_heater from homeassistant.components.water_heater
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.climate] Setting up climate.hive
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setting up sensor
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setup of domain sensor took 0.0 seconds
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setting up water_heater
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.setup] Setup of domain water_heater took 0.0 seconds
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.hive
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new binary_sensor.hive entity: binary_sensor.hive_hub_status
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.components.water_heater] Setting up water_heater.hive
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new climate.hive entity: climate.your_thermostat
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new sensor.hive entity: sensor.your_thermostat_battery_level
> 2021-02-15 20:54:07 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new water_heater.hive entity: water_heater.hot_water
> ```
> 
> Also tried turning on 2FA and that worked too. It prompted me to enter the number that had just appeared on my phone, it completed successfully, and bidirectional heating control worked fine.
> 
> So in summary it seems to work perfectly apart from the above error while installing the integration.

Thanks @davet2001 please can you add more specific details to this, when you say installation is this when you setup Hive via the UI? Or was your precious config imported from your configuration.yaml?

davet2001 commented:
> Thanks @davet2001 please can you add more specific details to this, when you say installation is this when you setup Hive via the UI? Or was your previous config imported from your configuration.yaml?

Hi, by installation, I mean:
1) freshly rebuilt dev container in vscode 
2) emptied config directory
3) start up HA in preview mode
4) on web interface click 'add integration' then select 'hive'

Error message occurs on step 4.
For my 2nd test (2FA) I repeated steps 2-4 but did not rebuild the dev container.

I can retry tonight in case I missed something in step 2 first time.

davet2001 commented:
This time it worked fine.

I did:
````
rm -rf config; mkdir config
rebuild container
run preview in vscode
wait for HA to start
go through onboarding process
configuration > integrations > add integration, hive
2FA process completes
````
No errors on the console.

Perhaps I did not properly empty my config directory first time.

So no more issues from my point of view. :)

Commit:
Added tests

Commit:
Added tests

Commit:
Added tests

Commit:
Added tests

KJonline commented:
@MartinHjelmare Latest fixes have been pushed to the branch

Commit:
Moved Services to Platform

Commit:
Moved Services to Platform

Commit:
Moved Services to Platform

Commit:
Moved Services to Platform

KJonline commented:
@MartinHjelmare Also as per the first pr I have moved the services to platform level.

Commit:
Updates from review

Commit:
Updates from review

Commit:
Updates from review

Commit:
Updates from review

Commit:
Updated Docs

Commit:
Updated Docs

Commit:
Updated Docs

Commit:
Fix unknown import

Commit:
Fix unknown import

Commit:
Fix unknown import

MartinHjelmare commented on homeassistant/components/hive/strings.json:
```diff
@@ -35,7 +35,8 @@
       "unknown": "Something has gone wrong please try again later.",
-      "unknown_entry": "Unable to find exisitng entry."
+      "unknown_entry": "Unable to find exisitng entry.",
+      "reauth_sucessfull": "Reauthentication has completed cuscessfully."
```
Use our common reference:
https://github.com/home-assistant/core/blob/04e07d8b2c3f2d65351d5606c7d7f770f59338a6/homeassistant/strings.json#L73

Commit:
Updated config flow

Commit:
Updated config flow

Commit:
Updated config flow

Commit:
Fixed test issue

Commit:
Fixed test issue

Commit:
Fixed test issue

bramkragten committed:
Add description to tts and notify services (#46764)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>

bramkragten committed:
Update homeassistant services.yaml (#46952)

Add names and missing target to turn off service

Commit:
Updated unknow entry

Commit:
Updated unknow entry

Commit:
Updated unknow entry

Commit:
Updated unknow entry

Commit:
Updated unknow entry

cgarwood committed:
Return states list from zwave_js get_config_parameters websocket if available (#46954)

spacegaier committed:
Add stop tilt support to KNX (#46947)

MartinHjelmare committed:
Catch more zwave_js errors (#46957)

raman325 committed:
Add zwave_js.set_config_parameter service (#46673)

* create zwave_js.set_config_value service

* update docstring

* PR comments

* make proposed changes

* handle providing a label for the new value

* fix docstring

* use new library function

* config param endpoint is always 0

* corresponding changes from upstream PR

* bug fixes and add tests

* create zwave_js.set_config_value service

* update docstring

* PR comments

* make proposed changes

* handle providing a label for the new value

* fix docstring

* use new library function

* config param endpoint is always 0

* corresponding changes from upstream PR

* bug fixes and add tests

* use lambda to avoid extra function

* add services description file

* bring back the missing selector

* move helper functions to helper file for reuse

* allow target selector for automation editor

* formatting

* fix service schema

* update docstrings

* raise error in service if call to set value is unsuccessful

* Update homeassistant/components/zwave_js/services.yaml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/zwave_js/services.yaml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/zwave_js/services.yaml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/zwave_js/services.yaml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/zwave_js/services.yaml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/zwave_js/services.yaml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* remove extra param to vol.Optional

* switch to set over list for nodes

* switch to set over list for nodes

Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bdraco committed:
Add missing target to increase_speed/decrease_speed service (#46939)

joncar committed:
LiteJet is now configured using config_flow (#44409)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Bump pymyq to fix myq in core (#46962)

jjlawren committed:
Fix Plex handling of clips (#46667)

MartinHjelmare committed:
Improve zwave_js config flow (#46906)

chemelli74 committed:
Fix Shelly mireds and color_temp return type (#46112)

bdraco committed:
Avoid having to ask for the bond token when possible during config (#46845)

farmio committed:
Fix KNX services.yaml (#46897)

mib1185 committed:
Implement additional DataUpdateCoordinator to harmonize the data update handling of Synology DSM (#46113)

vdheijden committed:
Add more shopping list services (#45591)

Co-authored-by: Franck Nijhof <git@frenck.dev>

tkdrob committed:
Use core constants for dht (#46029)

jjlawren committed:
Fix Plex showing removed shared users (#46971)

milanmeu committed:
Fix typo in fireservicerota strings (#46973)

raman325 committed:
Add zwave_js.refresh_value service (#46944)

* add poll_value service

* switch vol.All to vol.Schema

* more relevant log message

* switch service name to refresh_value, add parameter to refresh all watched values, fix tests

* rename parameter and create task for polling command so we don't wait for a response

* raise ValueError for unknown entity

* better error message

* fix test

raman325 committed:
deep copy zwave_js state in test fixtures so tests are more isolated (#46976)

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Automatically create HomeKit accessory mode entries (#46473)

When we set up HomeKit, we asked users if they wanted
to create an entry in bridge or accessory mode.

This approach required the user to understand how HomeKit works and
choose which type to create.

When the user includes the media player or camera domains,
we exclude them from the bridge and create the additional entries
for each entity in accessory mode.

farmio committed:
Update xknx to 0.17.1 (#46974)

marcelveldt committed:
Extend zwave_js discovery scheme for lights (#46907)

natekspencer committed:
Add switches and sensors to Litter-Robot (#46942)

allenporter committed:
Make FAN_ON use the max duration rather than 15 min default (#46489)

StevenLooman committed:
Handle device IP change in upnp (#46859)

frenck committed:
Add template support to service targets (#46977)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

farmio committed:
Validate KNX addresses (#46933)

raman325 committed:
Add ClimaCell weather integration (#36547)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MartinHjelmare committed:
Save mysensors gateway type in config entry (#46981)

meichthys committed:
Mullvad VPN (#44189)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>

MartinHjelmare committed:
Improve mysensors config flow (#46984)

bramkragten committed:
Fix TTS services name (#46988)

frenck committed:
Mullvad integration improvements (#46987)

frenck committed:
Merge branch 'master' into dev

bieniu committed:
Catch AuthRequired exception in confirm discovery step for Shelly config flow (#46135)

Noltari committed:
Restore Tado binary sensor attributes (#46069)

adrian-vlad committed:
Add enable and disable services for recorder (#45778)

raman325 committed:
Migrate zwave_js entities to use new unique ID format (#46979)

* migrate zwave_js entities to use new unique ID format

* remove extra param from helper

* add comment to remove migration logic in the future

* update comment

* use instance attribute instead of calling functino on every state update

MartinHjelmare committed:
Fix zwave_js config flow server version timeout (#46990)

fabaff committed:
Set awesomeversion to 21.2.3 (#46989)

ludeeus committed:
Add addon selector (#46789)

raman325 committed:
Climacell fixes: Use common keys for strings, fix temp_low measurement, add windy condition (#46991)

* use common keys for lat and long

* additional fixes

* Update homeassistant/components/climacell/strings.json

Co-authored-by: Milan Meulemans <milan.meulemans@live.be>

Co-authored-by: Milan Meulemans <milan.meulemans@live.be>

milanmeu committed:
Use location common key reference in totalconnect (#46995)

cgtobi committed:
Add Netatmo device trigger (#45387)

Co-authored-by: J. Nick Koston <nick@koston.org>

milanmeu committed:
Remove recursive key reference (#46999)

marcelveldt committed:
Add number platform to Z-Wave JS (#46956)

* add number platform to zwave_js integration
* add discovery scheme for thermostat valve control, using number platform

Co-authored-by: kpine <keith.pine@gmail.com>

raman325 committed:
Setup config entry even if vizio device is unreachable (#46864)

emontnemery committed:
Support value_template in MQTT triggers (#46891)

* Support value_template in MQTT triggers

* Rename value_template to payload_template

* Revert "Rename value_template to payload_template"

This reverts commit 902094eefc6612e6b5c3bdb7440520af050c7f20.

cgtobi committed:
Add Sonos media browser image proxy (#46902)

jjlawren committed:
Use Plex server URL as config entry title (#47010)

jjlawren committed:
Bump plexapi to 4.4.0 (#47007)

ntilley905 committed:
Add FAA Delays Integration (#41347)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

chemelli74 committed:
Add missing tilt icon to Shelly tilt sensor (#46993)

bramkragten committed:
Update frontend to 20210224.0 (#47013)

marcelveldt committed:
Change Z-Wave JS discovery logic to adopt changes to DeviceClass (#46983)

Co-authored-by: raman325 <7243222+raman325@users.noreply.github.com>

frenck committed:
Bump version to 2021.4.0dev0 (#47017)

balloob committed:
Convert discovery helper to use dispatcher (#47008)

homeassistant committed:
[ci skip] Translation update

fabaff committed:
Upgrade TwitterAPI to 2.6.8 (#47019)

starkillerOG committed:
Add Xiaomi Miio fan config flow (#46866)

* Miio fan config flow

* fix styling and imports

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* rename device -> entity

* fix indent

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

raman325 committed:
Fix zwave_js unique ID migration logic (#47031)

bdraco committed:
Ensure doorbird events are re-registered when changing options (#46860)

- Fixed the update listener not being unsubscribed

- DRY up some of the code

- Fix sync code being called in async

- Reduce executor jumps

balloob committed:
Remove deprecated credstash + keyring (#47033)

FHeilmann committed:
hm climate: Return PRESET_NONE instead of None (#47003)

Signed-off-by: Florian Heilmann <Florian.Heilmann@gmx.net>

chemelli74 committed:
Fix missing Shelly external input (#47028)

* Add support for external input (Shelly 1/1pm add-on)

* Make external sensor naming consistent

* Fix case consistency

Kane610 committed:
Normally there should only be one battery sensor per device from deCONZ. (#46761)

With these Danfoss devices each endpoint can report its own battery state.

fabaff committed:
Upgrade icmplib to 2.0.2 (#47039)

balloob committed:
Clean up discovery integration (#47022)

* Clean up discovery integration

* Fix tests

* Remove discovery step from freebox

fabaff committed:
Upgrade pyowm to 3.2.0 (#47042)

fabaff committed:
Upgrade sendgrid to 6.6.0 (#47041)

MartinHjelmare commented on homeassistant/components/hive/strings.json:
```diff
@@ -33,10 +33,10 @@
-      "already_configured": "Hive account is already configured.",
+      "already_configured": "[%key:common::config_flow::abort::already_configured_account%]",
       "unknown": "Something has gone wrong please try again later.",
       "unknown_entry": "Unable to find exisitng entry.",
```
```suggestion
      "unknown_entry": "Unable to find existing entry.",
```

MartinHjelmare commented on homeassistant/components/hive/config_flow.py:
```diff
@@ -106,7 +105,12 @@ async def async_step_finish(self):
+        schema = vol.Schema(
+            {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+        )
+        return self.async_show_form(step_id="user", data_schema=schema, errors=errors)
```
Don't move directly to another step's form. And don't pass errors between steps. Errors are specific to the form where they came from.

Since the finish step doesn't have a form, it's better to make it a helper method instead of a step method. Raise the errors encountered and catch them in the calling step method.

joogps committed:
Add remote control platform to Panasonic Viera (#42450)

* Adding remote platform

* Update homeassistant/components/panasonic_viera/remote.py

Simplify entity creation

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Use Pytest fixture

* Use Pytest fixtures and assert service calls

* Adding conftest.py and organizing tests

* Reorganizing tests

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bieniu committed:
Bump gios library to version 0.2.0 (#47050)

Melantrix committed:
Replace wrong domain returned from xbox api 2.0 (#47021)

* Change solution to use yarl lib

* Add check if base url needs changing

* Actively remove padding query instead of omitting

* Fixed popping the wrong query

* Change explaination about removing mode query

Quentame committed:
Add zeroconf discovery to Freebox (#47045)

* Add zeroconf discovery to Freebox

- remove deprecated discovery
- tried with SSDP too but the presentation URL is not the same (*.fbxos.fr for zeroconf, http://mafreebox.freebox.fr/ for SSDP)
- so config entry unique_id should be the MAC (included into SSDP, but not zeroconf, can be retrieve from `fbx.system.get_config()`)
- DHCP discovery might be added in the future too

* host and port are required on zeroconf

* cleanup in other PR

cgtobi committed:
Clean up Netatmo webhook handler (#47037)

cyberjunky committed:
Bump python-garminconnect to 0.1.19 to fix broken api (#47020)

Commit:
Change finish step into helper

Commit:
Change finish step into helper

Commit:
Change finish step into helper

Commit:
Change finish step into helper

Commit:
Change finish step into helper

bdraco committed:
Fix bond typing in config flow (#47055)

MartinHjelmare commented on homeassistant/components/hive/config_flow.py:
```diff
@@ -0,0 +1,177 @@
+class UnknownHiveError(Exception):
+    """Catch unknown hive error."""
+
+    pass
```
```suggestion
```

Commit:
Update homeassistant/components/hive/config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Commit:
Update homeassistant/components/hive/config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Commit:
Update homeassistant/components/hive/config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Commit:
Update homeassistant/components/hive/config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Commit:
Update homeassistant/components/hive/config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MartinHjelmare commented on homeassistant/components/hive/config_flow.py:
```diff
@@ -0,0 +1,175 @@
+        """Finish setup and create the config entry."""
+        self.data["tokens"] = self.tokens.get("AuthenticationResult")
+
+        if "AccessToken" in self.data["tokens"]:
+            # Setup the config entry
+            if self.entry:
+                self.hass.config_entries.async_update_entry(
+                    self.entry, title=self.data["username"], data=self.data
+                )
+                await self.hass.config_entries.async_reload(self.entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+            return self.async_create_entry(title=self.data["username"], data=self.data)
+
+        raise UnknownHiveError
```
```suggestion
        if "AccessToken" not in self.data["tokens"]:
            raise UnknownHiveError

        # Setup the config entry
        if self.entry:
            self.hass.config_entries.async_update_entry(
                self.entry, title=self.data["username"], data=self.data
            )
            await self.hass.config_entries.async_reload(self.entry.entry_id)
            return self.async_abort(reason="reauth_successful")
        return self.async_create_entry(title=self.data["username"], data=self.data)
```

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -52,101 +34,83 @@
-    hass.data[DOMAIN][DATA_HIVE] = hive
-    hass.data[DOMAIN]["entity_lookup"] = {}
+    if devices == "INVALID_REAUTH":
+        return hass.async_create_task(
```
We should always return False here. Create the task first, then return False.

MartinHjelmare commented on homeassistant/components/hive/strings.json:
```diff
@@ -0,0 +1,53 @@
+        "data": {
+          "username": "[%key:common::config_flow::data::username%]",
+          "password": "[%key:common::config_flow::data::password%]",
+          "scan_interval": "Scan Interval (seconds)"
```
There's no scan interval item in the user form data.

MartinHjelmare commented on homeassistant/components/hive/__init__.py:
```diff
@@ -52,101 +34,83 @@
 
-    hass.data[DOMAIN][DATA_HIVE] = hive
-    hass.data[DOMAIN]["entity_lookup"] = {}
+    if devices == "INVALID_REAUTH":
```
Side note: It's weird that `devices` both can be a string representing an error and a dict with hive types. I'd recommend instead to raise an exception when the authentication fails and we should reauthenticate. Return types should be as specific as possible.

ludeeus committed:
Use dispatch instead of eventbus for supervisor events (#46986)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bramkragten committed:
Updated frontend to 20210225.0 (#47059)

bramkragten committed:
Revert CORS changes for my home assistant (#47064)

* Revert CORS changes for my home assistant

* Update test_init.py

* Update test_init.py

homeassistant committed:
[ci skip] Translation update

raman325 committed:
catch ValueError when unique ID update fails because its taken and remove the duplicate entity (#47072)

cgarwood committed:
Fix Z-Wave JS API docstrings (#47061)

bdraco committed:
Ensure hue options show the defaults when the config options have not yet been saved (#47067)

balloob committed:
Bump Z-Wave JS Server Python to 0.20.0 (#47076)

balloob committed:
Upgrade aiohttp to 3.7.4 (#47077)

chpego committed:
Upgrade youtube_dl to version 2021.02.22 (#47078)

* Upgrade youtube_dl to version 2021.02.22

* Update requirements_all.txt

emontnemery committed:
Bump pychromecast to 8.1.2 (#47085)

Commit:
Add support for Shelly SHBTN-2 device triggers (#46644)

emontnemery committed:
Bump pychromecast to 9.0.0 (#47086)

* Adapt to Pychromecast 9.0.0

* Bump pychromecast to 9.0.0

* Fix lint issues

raman325 committed:
Remove flaky climacell test (#47080)

rikroe committed:
Bump bimmer_connected to 0.7.15 and fix bugs (#47066)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

Shutgun committed:
Add suggested_area support to devolo Home Control (#47063)

agners committed:
Add new machine generic-x86-64 to build matrix (#47095)

The Intel NUC machine runs on most UEFI capable x86-64 machines today.
Lets start a new machine generic-x86-64 which will replace intel-nuc
over time.

cgarwood committed:
Guard zwave_js missing nodes in websocket api (#47096)

bdraco committed:
Add suggested area to hue (#47056)

elupus committed:
Add support for v6 features to philips js integration (#46422)

chemelli74 committed:
Bump aioshelly to 0.6.1 (#47088)

marcelveldt committed:
Fix Z-Wave JS discovery schema for thermostat devices (#47087)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

allenporter committed:
Bump google-nest-sdm to v0.2.12 to improve API call error messages (#47108)

balloob committed:
Use async_capture_events to avoid running in executor (#47111)

thecode committed:
Fix Shelly RGBW (#47116)

homeassistant committed:
[ci skip] Translation update

bramkragten committed:
Update frontend to 20210226.0 (#47123)

bdraco committed:
Provide a human readable exception for the percentage util (#47121)

bdraco committed:
Handle lutron_caseta fan speed being none (#47120)

allenporter committed:
Improve handling for recording start of nest cam stream (#47144)

* Improve handling for start of nest cam stream

Add negative_cts_offsets to segment container options in order to better handle recording at the start of a stream. Nest streams start off with a negative offset, and if the segment container does not support it, then it adjusts the timestamps making it out of order with the next segment as described in issue #46968

* Update homeassistant/components/stream/__init__.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

Kane610 committed:
Change device class of window covers to shade (#47129)

homeassistant committed:
[ci skip] Translation update

bieniu committed:
Bump airly library to version 1.1.0 (#47163)

tkdrob committed:
Clean up mqtt_room (#46882)

agners committed:
Bump builder to get generic-x86-64 nightly builds (#47164)

balloob committed:
Fix the updater schema (#47128)

dmulcahey committed:
Bump ZHA quirks to 0.0.54 (#47172)

emontnemery committed:
Tweak Tasmota fan typing (#47175)

mib1185 committed:
Apply recommendations to synology_dsm (#47178)

allenporter committed:
Fix flaky hls keepalive test (#47186)

Remove a call to stream.start() which is issued before the test is fully
setup (e.g. keepalive is not set to True, and mock calls are not registered)

bdraco committed:
Update HAP-python to 3.3.1 (#47180)

Fixes disconnect when setting a single char fails
https://github.com/ikalchev/HAP-python/compare/v3.3.0...v3.3.1

emontnemery committed:
Fix MQTT trigger where wanted payload may be parsed as an integer (#47162)

emontnemery committed:
Update MQTT device triggers with support for templates (#47142)

starkillerOG committed:
Fix Xiaomi Miio discovery (#47134)

ajschmidt8 committed:
Update AlarmDecoder dependency (#46841)

mdz committed:
Remove turn_on and turn_off from SmartTub pump switches (#47184)

mdz committed:
Move SmartTub climate constants to module level (#47190)

esev committed:
Explain why should_pool is True initially for wemo (#47191)

mdz committed:
Cleanup SmartTub filtration cycles (#47192)

homeassistant committed:
[ci skip] Translation update

allenporter committed:
Handle stream failures in recorder (#47151)

* Handle stream failures in recorder

Fail gracefully with an error message when the recorder is invoked with no segments due to a stream failure.

* Update homeassistant/components/stream/recorder.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

bdraco committed:
Enforce typing in bond (#47187)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bachya committed:
Bump simplisafe-python to 9.6.7 (#47206)

bdraco committed:
Increment the homekit config version when restarting (#47209)

If an entity changes between restart the iOS/controller
device may have cached the old chars for the accessory.
To force the iOS/controller to reload the chars, we
increment the config version when Home Assistant restarts

bdraco committed:
Bump HAP-python to 3.3.2 to fix unavailable condition on restart (#47213)

Fixes

https://github.com/ikalchev/HAP-python/compare/v3.3.1...v3.3.2

cdce8p committed:
Update pylint (#47205)

agners committed:
Fix generic-x86-64 build (#47214)

Replace the wrong Docker Hub repository slipped in during testing.

raman325 committed:
Add hassio addon_update service and hassio config entry with addon and OS devices and entities (#46342)

* add addon_update service, use config flow to set up config entry, create disabled sensors

* move most of entity logic to common entity class, improve device info, get rid of config_flow user step

* fix setup logic

* additional refactor

* fix refactored logic

* fix config flow tests

* add test for addon_update service and get_addons_info

* add entry setup and unload test and fix update coordinator

* handle if entry setup calls unload

* return nothing for coordinator if entry is being reloaded because coordinator will get recreated anyway

* remove entry when HA instance is no longer hassio and add corresponding test

* handle adding and removing device registry entries

* better config entry reload logic

* fix comment

* bugfix

* fix flake error

* switch pass to return

* use repository attribute for model and fallback to url

* use custom 'system' source since hassio source is misleading

* Update homeassistant/components/hassio/entity.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* update remove addons function name

* Update homeassistant/components/hassio/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* fix import

* pop coordinator after unload

* additional fixes

* always pass in sensor name when creating entity

* prefix one more function with async and fix tests

* use supervisor info for addons since list is already filtered on what's installed

* remove unused service

* update sensor names

* remove added handler function

* use walrus

* add OS device and sensors

* fix

* re-add addon_update service schema

* add more test coverage and exclude entities from tests

* check if instance is using hass OS in order to create OS entities

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

cdce8p committed:
Improve CI workflow (#46696)

cgtobi committed:
Revert const replacement in fritzbox_callmonitor (#47211)

bdraco committed:
Remove griddy integration (#47218)

maxwroc committed:
Fix number of reported issues by github integration (#47203)

farmio committed:
KNX services send and event_register accept multiple group addresses (#46908)

* send and event_register service accept lists of group addresses

* remove lambda

* object selector for lists

* knx.read takes lists too

farmio committed:
Deprecate knx config_file (#46874)

* deprecate config_file

* removed cv.deprecated for now, added persistent notification

* Update homeassistant/components/knx/__init__.py

Co-authored-by: Philip Allgaier <philip.allgaier@gmx.de>

* remove notification, add cv.deprecated again

* Update homeassistant/components/knx/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* remove cv.deprecated again

Co-authored-by: Philip Allgaier <philip.allgaier@gmx.de>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

emontnemery committed:
Fix race when disabling config entries (#47210)

* Fix race when disabling config entries

* Remove unused constant

starkillerOG committed:
Fix Xiaomi Miio flow unique_id for non discovery flows (#47222)

frenck committed:
Restore pylint concurrency (#47221)

mdz committed:
Parameterize SmartTub tests (#47189)

* Parameterize SmartTub tests

* parameterize light service calls

* remove stray print()

* add comment

frenck committed:
Upgrade coverage to 5.5 (#47227)

frenck committed:
Upgrade spotipy to 2.17.1 (#47228)

jdeath committed:
Bump mcstatus to 5.1.1 (#47169)

dermotduffy committed:
Overhaul command_line tests (#46682)

thecode committed:
Fix Shelly Polling (#47224)

Commit:
Merge branch 'Hive_Config-Flow' of https://github.com/Pyhive/core into Hive_Config-Flow

* 'Hive_Config-Flow' of https://github.com/Pyhive/core:
  Update homeassistant/components/hive/config_flow.py
  Change finish step into helper
  Updated unknow entry
  Fixed test issue
  Updated config flow
  Fix unknown import
  Updated Docs
  Updates from review
  Moved Services to Platform
  Added tests
  Updated Strings.json
  update to Bool check
  hive config flow

dermotduffy committed:
Minor Hyperion mypy cleanups (#45765)

raman325 committed:
Address late hassio review (#47229)

* hassio code cleanup to address comments in #46342

* fix code

farmio committed:
KNX address constant (#47196)

emontnemery committed:
Pass variables to initial evaluation of template trigger (#47236)

* Pass variables to initial evaluation of template trigger

* Add test

* Clarify test

emontnemery committed:
Add support for a list of known hosts to Google Cast (#47232)

bdraco committed:
Fix lutron caseta fan handling of speed off (#47244)

elupus committed:
Add remote control support to philips_js (#47249)

raman325 committed:
Add suggested area for zwave_js devices (#47250)

bramkragten committed:
Update frontend to 20210301.0 (#47252)

bachya committed:
Bump simplisafe-python to 9.6.8 (#47241)

balloob committed:
Revert "Fix the updater schema (#47128)" (#47254)

This reverts commit 98be703d90e44efe43b1a17c7e5243e5097b00b1.

frenck committed:
Fix duplicate template handling in Persistent Notifications (#47217)

frenck committed:
Deprecate LIFX Legacy integration (#47235)

milanmeu committed:
Fix typo in plaato strings (#47245)

bdraco committed:
Fix harmony failing to switch activities when a switch is in progress (#47212)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

stephan192 committed:
Remove rounding from The Things Network (#47157)

marcelveldt committed:
Update color logic for zwave_js light platform (#47110)

Co-authored-by: Raman Gupta <7243222+raman325@users.noreply.github.com>

davet2001 commented:
Rebase maybe? Diff shows 25k lines for me currently.

emontnemery committed:
Add battery sensor for gogogate2 wireless door sensor (#47145)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Fix typing on fan percentage (#47259)

KJonline commented:
> Rebase maybe? Diff shows 25k lines for me currently.

Ye thanks for spotting it for some reason when I rebase it causes this and then I have to rebase a second time to fix it. Thanks for letting me know.

scop committed:
Lint suppression cleanups (#47248)

* Unused pylint suppression cleanups

* Remove outdated pylint bug references

* Add flake8-noqa config and note to run it every now and then

* Add codes to noqa's

* Unused noqa cleanups

tkdrob committed:
Clean up constants (#46948)

* Clean up constants

* clean up humidifier constants

* fix tests

* fix prometheus tests

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

Quentame committed:
Add init test to Freebox (#46998)

* Add init test to Freebox

* Review : more readable conftest

* Expect 2 blank lines between defs

* Review : Not I/O in the event loop

* Fix test_setup test

* remove useless const

* Review : mock setup methods

* Add service test

* Add import test

Nick-Adams-AU committed:
Add services for izone airflow min/max (#45727)

* Create airflow_min and airflow_max services for the izone component
* Bump pizone library requirement

emontnemery committed:
Make MQTT number respect retain setting (#47270)

balloob committed:
Fix Alexa doorbells (#47257)

frenck committed:
Fix izone flake8 error (#47276)

davet2001 commented:
Not sure whether the prob is my end but still shows 25k diff.

Anyway I have checked out your latest and tried it on my live hive system - all still working fine.

Hoping this can make it into the March release.

rlehfeld committed:
Add force_update to tasmota sensors (#47052)

* Add force update also to non-binary sensors as e.g. POWER Measurement agerage cannot be calculated otherwise. This is the same behavior as set with the obsolete tasmota detection

* add tests in binary_sensor and test_sensor for force_update flag

* satisfy flake8

* next try for force_update test but this time on the entity object which is the correct level

* once again satisfy flake8

* one more try for a test

* fix typo

* satisfy black

bramkragten committed:
Update frontend to 20210302.0 (#47278)

bachya committed:
Bump simplisafe-python to 9.6.9 (#47273)

Adminiuga committed:
Update ZHA dependencies (#47282)

scop committed:
Upgrade isort to 5.7.0 (#47279)

https://pycqa.github.io/isort/CHANGELOG/#570-december-30th-2020

Quentame committed:
Add disk sensor to Freebox (#46689)

* Add disk sensor to Freebox

* Add debug logging into sensors

* Remove useless sensor[X] assignement in disk sensor

Quentame committed:
Uniformize platform setup (#47101)

* A platform is not a component

* Fix dynalite

* SUPPORTED_PLATFORMS --> PLATFORMS

* In tests

* In tests 2

* Fix SmartThings

* Fix ZHA test

* Fix Z-Wave

* Revert Z-Wave

* Use PLATFORMS const in ambient_station

* Fix ihc comment

chemelli74 committed:
Limit log spam by ConfigEntryNotReady (#47201)

balloob committed:
Clean up secret loading (#47034)

frenck committed:
Upgrade pillow to 8.1.1 (#47223)

cgtobi committed:
Fix issue when setting boost preset for a turned off Netatmo thermostat (#47275)

raman325 committed:
Add raw values to zwave_js value notification event (#47258)

* add value_raw to value notification event that always shows the untranslated state value

* add property key and property to event params

raman325 committed:
Convert climacell forecast timestamp to isoformat so that UI shows the right times (#47286)

raman325 committed:
Bump zwave-js-server-python to 0.20.1 (#47289)

MartinHjelmare committed:
Add zwave_js add-on manager (#47251)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

Commit:
Next round of fixes

Commit:
Next round of fixes

Commit:
Next round of fixes

Commit:
Merge branch 'Hive_Config-Flow' of https://github.com/Pyhive/core into Hive_Config-Flow

* 'Hive_Config-Flow' of https://github.com/Pyhive/core: (27 commits)
  Next round of fixes
  Update homeassistant/components/hive/config_flow.py
  Change finish step into helper
  Updated unknow entry
  Fixed test issue
  Updated config flow
  Fix unknown import
  Updated Docs
  Updates from review
  Moved Services to Platform
  Added tests
  Updated Strings.json
  update to Bool check
  hive config flow
  Update homeassistant/components/hive/config_flow.py
  Change finish step into helper
  Updated unknow entry
  Fixed test issue
  Updated config flow
  Fix unknown import
  ...

Commit:
Merge branch 'Hive_Config-Flow' of https://github.com/Pyhive/core into Hive_Config-Flow

* 'Hive_Config-Flow' of https://github.com/Pyhive/core: (27 commits)
  Next round of fixes
  Update homeassistant/components/hive/config_flow.py
  Change finish step into helper
  Updated unknow entry
  Fixed test issue
  Updated config flow
  Fix unknown import
  Updated Docs
  Updates from review
  Moved Services to Platform
  Added tests
  Updated Strings.json
  update to Bool check
  hive config flow
  Update homeassistant/components/hive/config_flow.py
  Change finish step into helper
  Updated unknow entry
  Fixed test issue
  Updated config flow
  Fix unknown import
  ...

KJonline commented:
@MartinHjelmare Something seems to have gone wrong with this brand I have tried rebasing 3 times and I cannot seem to remove the tainted commits. I have created a replacement PR in #47300 to continue the review.

@davet2001  please can you test the code again on the that branch for sanity? 

Thanks
Khole
